edb/lib/ext/auth.edgeql

```diff
@@ -135,6 +135,44 @@ CREATE EXTENSION PACKAGE auth VERSION '1.0' {
         };
     };
 
+    create abstract type ext::auth::Event extending std::BaseObject {
+        create required property timestamp: std::datetime {
+            set default := std::datetime_current();
+            set readonly := true;
+        };
+    };
+
+    create scalar type ext::auth::IdentityEventType extending enum<
+        Created,
+        Deleted,
+        Authenticated,
+    >;
+
+    create abstract type ext::auth::IdentityEvent extending ext::auth::Event {
+        create required property event_type: ext::auth::IdentityEventType;
+
+        create required link identity: ext::auth::Identity {
+            on target delete delete source;
+        };
+    };
```

Comment:
How about making `identity` link optional and moving it into `ext::auth::Event`? IMO for logging kind of types it's more convenient to "compress" all relevant links/properties into just one type.

Reply:
There is also the `token` property for some of the events (needed for sending an email with a reset or verification token, for instance). We could also drop that entirely and make it _just_ for the email/webhook and not store it in the event, but I kind of wanted the event to be the source of truth here. But, that was also before the net module existed which has a record of everything anyway.